src/Shared/Targets/Wrappers/AspNetBufferingTargetWrapper.cs

```diff
@@ -0,0 +1,305 @@
+            }
+            InternalLogger.Trace("Sending buffered events to wrapped target: {0}.", WrappedTarget);
+            WrappedTarget?.WriteAsyncLogEvents(buffer.GetEventsAndClear());
+        }
+
+        /// <summary>
+        /// Obtains a slot in the buffer dictionary for 'this' class instance
+        /// If that does not exist, that is created first
+        /// if the dictionary does not exist, that is created first
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        private Internal.LogEventInfoBuffer GetOrCreateRequestBuffer(HttpContextBase context) 
+        {
+            // If the context is missing, stop
+            if (context == null)
+            {
+                return null;
+            }
+
+            var bufferDictionary = GetBufferDictionary(context);
+
+            // If the dictionary is missing, stop
+            if (bufferDictionary == null)
+            {
+                return null;
+            }
+
+            lock (bufferDictionary)
+            {
+                if (!bufferDictionary.TryGetValue(this, out var buffer))
+                {
+                    buffer = new Internal.LogEventInfoBuffer(BufferSize, GrowBufferAsNeeded, BufferGrowLimit);
+                    bufferDictionary.Add(this, buffer);
+                }
+
+                return buffer;
+            }
+        }
+
+        /// <summary>
+        /// The Key for the buffer dictionary in the HttpContext.Items collection
+        /// </summary>
+        private static readonly object HttpContextItemsKey = new object();
+
+        /// <summary>
+        /// Return the buffer dictionary from the HttpContext.Items using HttpContextItemsKey
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        private static Dictionary<AspNetBufferingTargetWrapper, Internal.LogEventInfoBuffer> GetBufferDictionary(HttpContextBase context)
+        {
+            return context?.Items?[HttpContextItemsKey] as
+                Dictionary<AspNetBufferingTargetWrapper, Internal.LogEventInfoBuffer>;
+        }
+
+        /// <summary>
+        /// Create the buffer dictionary in the HttpContext.Items using HttpContextItemsKey
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        private static void SetBufferDictionary(HttpContextBase context)
+        {
+            context.Items[HttpContextItemsKey] = new Dictionary<AspNetBufferingTargetWrapper, Internal.LogEventInfoBuffer>();
+        }
+
+        /// <summary>
+        /// Adds the specified log event to the buffer.
+        /// NOTE: if Write is never called, this instance will not be registered in the buffer dictionary in HttpContext.Items.
+        /// That is expected normal behavior.
+        /// </summary>
+        /// <param name="logEvent">The log event.</param>
+        protected override void Write(AsyncLogEventInfo logEvent)
```

Comment:
Can this method be moved up to be above `private void Flush(Internal.LogEventInfoBuffer buffer)`

Again to reduce the changes to make the rename automatic.